### DISCUSSION TITLE
Is there a way to add buildModules via the moduleContainer class?

### DISCUSSION THREAD

**Discussion by lopugit** [nuxt/nuxt#10247] [Questions (Nuxt 2)]
We currently have a flow where we use 1 parent module to extend a lot of our sites nuxt config in the same way, calling this.addModule() a lot inside its main exported function of the module.

I was wondering if there's a way to extend and add build modules via another module?

Kindly

**Comment by weotch** [comment]
Same question/issue, same use case.  [For example](https://github.com/BKWLD/cloak-boilerplate/blob/f0af37145bd07305f505568c3d78c3f005e0a9fa/modules/style.js#L28-L29).

**Comment by danielroe** [comment]
A buildModule is only different by being loaded purely in development. So you can do:

```js
if (this.nuxt.options.dev) {
  this.addModule(/* */)
}
```

**Reply by weotch** [reply]
When running `nuxt generate`, this option is `false`.  However, I do want these modules present during the build phase of SSG.  Should I just not be trying to use `buildModules` for things required by the build phase of SSG?  Or is there a different signal I should use?  Thanks!

**Reply by pi0** [reply]
Check for options._build it covers all 3 cases

**Comment by pencilcheck** [comment]
Didn't work for @nuxtjs/tailwindcss, when I do it, it errors out

```
  if (this.nuxt.options.dev) {
    console.log('this.nuxt.options.dev', this.nuxt.options.dev)
    this.requireModule('@nuxt/typescript-build')
    this.requireModule('@nuxtjs/tailwindcss')
    this.requireModule('@nuxtjs/google-fonts')
    this.requireModule('@nuxtjs/style-resources')
    this.requireModule('@nuxtjs/device')
    this.requireModule('@nuxtjs/fontawesome')
    this.requireModule('@braid/vue-formulate/nuxt')
    this.requireModule('@nuxtjs/composition-api/module')
    this.requireModule('@pinia/nuxt')
  }
```

```
 ERROR  input.includes is not a function

  at normalizeWindowsPath (/project/node_modules/pathe/dist/shared/pathe.f81973bb.mjs:2:24)
  at normalize (/project/node_modules/pathe/dist/shared/pathe.f81973bb.mjs:17:10)
  at resolvePath (/project/node_modules/@nuxt/kit/dist/index.mjs:330:10)
  at /project/node_modules/@nuxtjs/tailwindcss/dist/module.mjs:167:75
  at Array.map (<anonymous>)
  at setup (/project/node_modules/@nuxtjs/tailwindcss/dist/module.mjs:167:64)
  at async ModuleContainer.normalizedModule (/project/node_modules/@nuxt/kit/dist/index.mjs:637:5)
  at async ModuleContainer.addModule (/project/node_modules/@nuxt/core/dist/core.js:239:20)
```

hmm, it could be related to this bug: https://github.com/nuxt-modules/tailwindcss/issues/505

But it works if I simply include those buildModules in nuxt.config instead, why??


Nvm, looks like one of the library injected an object into the css array for some reason....
